relay-metrics/src/aggregation.rs

```diff
@@ -1717,7 +1733,13 @@ impl Aggregator {
                     cost_tracker.subtract_cost(key.project_key, key.cost());
                     cost_tracker.subtract_cost(key.project_key, value.cost());
                     let bucket = Bucket::from_parts(key.clone(), bucket_interval, value);
-                    buckets.entry(key.project_key).or_default().push(bucket);
+                    buckets
+                        .entry(key.project_key)
+                        .or_default()
+                        .push(HashedBucket {
+                            hashed_key: key.as_integer_lossy(), // TODO: Do we need a more reliable hasher?
```

Comment:
Not sure if we can rely on this hasher, this comment seems pretty clear:

https://github.com/getsentry/relay/blob/a9d93903f11112ed4d3222646df84c0c9fe24d74/relay-metrics/src/aggregation.rs#L894-L895

Reply:
I think I looked into this, didn't find any great, fast crates, and then @jan-auer pointed out that we already have a hasher for this.

Considering we're doing sharding of kafka topics by org id at some point, I think it's time well spent to look into a hashing function that is deterministic and portable (can be replicated in Python), and I am not sure if this function here fulfills any of those criteria.

Reply:
> we already have FMV for this.

Good point, will replace with FnvHasher.

Reply:
There's a Python impl for FNV, and it looks simple enough to write ourselves if necessary:
https://pypi.org/project/fnvhash/
https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function